src/common-utils/main.sh

```diff
@@ -437,6 +437,14 @@ if [ "${INSTALL_ZSH}" = "true" ]; then
     fi
 
     if [ "${CONFIGURE_ZSH_AS_DEFAULT_SHELL}" == "true" ]; then
+        # Fixing chsh always asking for a password on alpine linux
+        # ref: https://askubuntu.com/questions/812420/chsh-always-asking-a-password-and-get-pam-authentication-failure.
+        if [ -f "/etc/pam.d/chsh" ] && grep -Eq "^auth(.*)pam_rootok\.so$" /etc/pam.d/chsh; then
+            awk '/^auth(.*)pam_rootok\.so$/ { $2 = "sufficient" } { print }' /etc/pam.d/chsh > /tmp/chsh.tmp && mv /tmp/chsh.tmp /etc/pam.d/chsh
```

Comment:
Suggestion: Can add another condition so that we could avoid running this command when `/etc/pam.d/chsh` already contains `auth sufficient pam_rootok.so` ?


Something like 👇 

```

if [ -f "/etc/pam.d/chsh" ]; then
    if grep -vEq "^auth sufficient pam_rootok.so$"; then
            awk '/^auth(.*)pam_rootok\.so$/ { $2 = "sufficient" } { print }' /etc/pam.d/chsh > /tmp/chsh.tmp && mv /tmp/chsh.tmp /etc/pam.d/chsh
   fi
else
...

```

Reply:
Looks like there's a bug, I tested with ubuntu where the `/etc/pam.d/chsh` file exists. It is still trying to add `auth sufficient pam_rootok.so`. Also, in this case, it is replacing the contents - which doesn't look correct

![image](https://github.com/devcontainers/features/assets/24955913/7b7ab029-9b3a-42b3-ac18-86ee35363431)

Reply:
With https://github.com/devcontainers/features/pull/557/commits/0ae570c72222c749911dda57af6962b8bca86187, instead of replacing contents, it is now appending to the end

![image](https://github.com/devcontainers/features/assets/24955913/f096a202-6a59-433e-906d-7811e3908874)

Reply:
I'm not sure how to reproduce this. Could you give me some guidance/instructions? Thx